Add lifecycle status field with allowed values

brylie opened the pull request:
Closes #75 

# Changes
- add `lifecycleStatus` field to `ApiSchema`
  - add allowed values:
    - development
    - deprecated
    - design
    - production
    - testing
- Add Lifecycle Status field to **Add API** form
- label on API Profile page
- show different bootstrap label color based on status
- inline edit
- add internationalization

bajiat commented:
@marla-singer Could you review this PR? We would like to get it for release today, if possible.

marla-singer commented:
Reviewing

marla-singer commented:
@brylie As for me life status label looks bad and very big in header (Chrome browser)
![joxi_screenshot_1482223864162](https://cloud.githubusercontent.com/assets/20643403/21344089/0f41226a-c6ab-11e6-9bf4-e18cd5de7e3c.png)

marla-singer commented:
> Owner can add the lifecycle status in Add form
Didn't find field to add status in Add from

marla-singer commented:
@brylie Lifecycle status doesn't appear on catalog card? 
![joxi_screenshot_1482225113563](https://cloud.githubusercontent.com/assets/20643403/21344625/c737671a-c6ad-11e6-8a10-88b36efc9fdd.png)

marla-singer commented on apis/client/view/lifecycle_status/lifecycle_status.js:
```diff
@@ -0,0 +1,93 @@
+    return statusText;
+  },
+  lifecycleStatusField () {
+    return ['lifecycleStatus'];
```
Do we really need this helper?

brylie replied:
The helper is necessary because the quickForm expects an array of options. We cannot pass in an array from the Handlebars template.

marla-singer replied:
Then add comment that describes this helper

brylie commented:
> Lifecycle status doesn't appear on catalog card?

That will be added in issue #76

brylie commented:
> Didn't find field to add status in Add from

The field has now been added.

brylie commented:
> As for me life status label looks bad and very big in header 

Fixed.
![screenshot_20161220_122901](https://cloud.githubusercontent.com/assets/17307/21347110/f316e8ae-c6af-11e6-8186-c6ca8dd1c013.png)

marla-singer commented:
Test is with Private label
![joxi_screenshot_1482229809023](https://cloud.githubusercontent.com/assets/20643403/21347151/79db2b18-c6b8-11e6-9fd3-11f6eecac509.png)

brylie commented:
> When the status is changed from "Undefined", Undefined should not appear anymore from the status selector dropdown.

I am not sure that this should be a requirement. By default, quick form has the 'select an API status' as the first option, which allows for the 'undefined' status. I don't think it is justified to fight the default behavior here, since this is not clearly causing any problems.

brylie commented on apis/client/view/lifecycle_status/lifecycle_status.html:
```diff
@@ -0,0 +1,31 @@
+      {{/ unless }}
+
+      <!-- edit mode form -->
+      {{# if editMode }}
```
Done.

brylie commented:
> Test is with Private label
> ![private label UI](https://cloud.githubusercontent.com/assets/20643403/21347151/79db2b18-c6b8-11e6-9fd3-11f6eecac509.png)

Fixed:

![screenshot_20161220_130017](https://cloud.githubusercontent.com/assets/17307/21348112/51098a62-c6b4-11e6-967a-1503c17bddfb.png)

I removed the inline edit, since we have so much complication here.

marla-singer commented on apis/client/view/lifecycle_status/lifecycle_status.js:
```diff
@@ -0,0 +1,58 @@
+
+    if (templateInstance.data.api && templateInstance.data.api.lifecycleStatus) {
+      // Get lifecycle status from API document
+      status = templateInstance.data.api.lifecycleStatus;
```
Forget add `const` for variable

marla-singer commented on apis/client/view/lifecycle_status/lifecycle_status.js:
```diff
@@ -0,0 +1,58 @@
+      // Get translation text for Unknown lifecycle status
+      const unknown = TAPi18n.__('apiLifecycleStatus_labelText_unknown');
+
+      statusText = unknown;
```
Local variable `unknown` is redundant. Please leave it as
```javascript
statusText = TAPi18n.__('apiLifecycleStatus_labelText_unknown');
```

marla-singer commented:
> I added an &nbsp; between the labels for spacing, because I could not figure out why our label spans were not behaving like the Bootstrap documentation indicates:

It happens because of i18n tag. If these is to leave just "Private" then it works as expected

brylie commented:
Alright, well I am cool having an `&nbsp;` line there, if it means we can localize the 'private' label.

marla-singer commented:
@brylie Space symbol is ok
